System/Process.hs

```diff
@@ -870,6 +869,22 @@ It will therefore behave more portably between operating systems than 'system'.
 The return codes and possible failures are the same as for 'system'.
 -}
 rawSystem :: String -> [String] -> IO ExitCode
-rawSystem cmd args = do
-  (_,_,_,p) <- createProcess_ "rawSystem" (proc cmd args) { delegate_ctlc = True }
+rawSystem cmd args = executeAndWait "rawSystem" (proc cmd args) { delegate_ctlc = True }
+
+-- ---------------------------------------------------------------------------
+-- executeAndWait
+
+-- | Create a new process and wait for it's termination.
```

Comment:
Please explain how this is different from `createProcess` followed by `waitForProcess`.  It's quite worrying to have two APIs for the same thing that have subtly different behaviour, but I didn't follow all the details of the problem.  Is there really no way to make `waitForProcess` do the right thing?